specs/fjord/derivation.md

```diff
@@ -0,0 +1,20 @@
+# L2 Chain Derivation
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Architecture](#architecture)
+  - [L2 Chain Derivation Pipeline](#l2-chain-derivation-pipeline)
+    - [Batch Queue](#batch-queue)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Architecture
+
+## L2 Chain Derivation Pipeline
+
+### Batch Queue
+
```

Comment:
This seems like it would be perfect for a constants table. It would also be amazing to include information like explicitly saying it was configurable before, tradeoffs of smaller/larger values and why this value was chosen

Reply:
@tynes I wasn't sure how much background info we actually want to give in the spec, so I rather started short to only state the most relevant information to implementers of what actually changes. But I'm happy to add this information if we want to give more context for the change.

Regarding a `constants` table, we don't have that yet in the main spec (`protocol/` folder). Where would you add this table? If I add it here, it would be a table with only a single entry.

Having said that, we might want to tweak two further constants in Fjord, namely `MaxChannelBankSize` and `MaxRLPBytesPerChannel`. @trianglesphere has requested to change these as well.

Reply:
This highlights the problem of defining what a good spec is and scaling this information. I personally lean towards including rationale in the spec so that we can scale the knowledge, which is why i always put "security considerations" sections in specs that i work on

Reply:
We don't yet have a constants table but its something we have been slowly moving towards :) curious how you feel about it, I think a table with 1 value is fine for now so it can be added to in the future

Reply:
@tynes I agree, it's a good idea to but more context into the specs. It makes them easier to understand! Added two sections on rationale and security considerations.

I also added tables to the Fjord derivation spec, as well as added one to the main derivation spec. I believe it's nice for implementers of the spec to have a table of protocol parameters and constants that are affected by protocol upgrades at least.